### PR TITLE
DeepLPF: training updates and single-image inference

### PR DESCRIPTION
**Pull request by sjmoran**
This update fixes DeepLPF filter training and scaling behaviour, adds single-image inference, and makes the implementation easier to install and maintain. I am the paper's first author.

### Changes

- Refine filter training with differentiable binarisation and branch selection, shared ellipse geometry across colour channels, and trainable graduated-filter line geometry.
- Fuse scaling maps as `1 + (s_g - 1) + (s_e - 1)`, so two neutral filters leave the image unchanged. Correct the MS-SSIM calculation so its top-scale term is applied once.
- Split the model and training code into focused modules while retaining existing import names. Remove unused modules that prevented strict loading of the supplied checkpoint.
- Add `pip install -e .` and `deeplpf enhance photo.jpg`, dependency updates, and inference for images without reference targets.
- Include updated documentation, examples, and a retrained checkpoint reported at **24.18 dB PSNR / 0.917 SSIM** on 500 FiveK test images.

Optional filter-count, colour-head, batching, and acceleration features are disabled by default. Implementation changes are under `DeepLPF/`; the diff also includes the root-level PR drafts `PULL_REQUEST.md` and `PULL_REQUEST_long.md`. The license file is unchanged; headers and README are aligned with its MIT license.

### Validation

Reported validation on CPU with Python 3.11, PyTorch 2.12, scikit-image 0.26, and NumPy 2.4:

- **45 tests passed**, including nine covering model and loss changes.
- Refactor and batching checks preserved forward outputs and loss, with maximum gradient difference `3.7e-09`.
- A synthetic-data run covered training, checkpointing, evaluation, and inference without reference images.

To run the tests without a dataset:

```bash
cd DeepLPF
pip install -r requirements.txt
python -m pytest test_deeplpf.py
```

The checkpoint result is from one 1,000-epoch run, without per-change ablations. GPU acceleration options have not been exercised, and the optional model features have not demonstrated an accuracy improvement. The model and loss fixes change training behaviour; this is not solely a refactor.

### CODE REVIEW

**Comment by sjmoran** [comment]
Hi @huawei-noah-admin, just following up on this DeepLPF update. I noticed this account has handled recent merges in the repository, so I hoped you could help route this review.

I am the paper’s first author and would be happy to address any questions or requested changes. Could you review this PR or point me to the current DeepLPF maintainer?

Thanks!
